keymanagementrequirements.md

```diff
@@ -0,0 +1,138 @@
+- Signing key: A signing key is used to generate artifact signatures. A signing key should be signed with a root key or one of its intermediaries. The certificate chain with a signing key can be used to verify which root it belongs to. While a root key can be used as a signing key, this is not recommended as it creates a large blast radius and increases the risk of compromising a root key. 
+- Trust Store: The trust store defines the relationship between signing keys and artifacts that are used at validation time to determine whether to trust an artifact with a crytptographically valid signature. The trust store will relate a source (any source, specific registry, specific repository, or specific target) with a certificate (for root key, intermediate, or signing key) or key repository (for automated key distribtuion). It is not recommended to use a signing key as this will cause signature validation to fail if the signing key is rotated. 
+- Trust Store Key Information: The key information configured in the trust store will be cryptographically verifiable and contain:
+    - Public Key: Will be used to verify signed artifacts.
+    - Signed Identity: Identity of creator of the signing key.
+    - (Optional) Issuer: If the key is an intermediate this will describe the owner of the root.
+    - Mechanism to update validity of signatures generated with the signing key.
+
+## Requirements
+- Signing an artifact MUST NOT require the publisher to perform additional actions with a registry/repository or registry operator. Uploading a signature MAY require additional actions.
+- Retrieving a signature MUST NOT require the deployer to perform additional actions with a registry/repository or registry operator beyond those required to pull an unsigned artifact.
+- Artifact integrity, source, and signature expiry MUST be verifiable from the signature AND NOT require additional calls to the registry/repository.
+- Signature allow list/deny list MAY require additional calls that will be defined in a separate document.
+- Moving an artifact from one repository to another SHOULD NOT invalidate the signature on the artifact.
+- Publishers SHOULD be able to sign with keys stored on their local machines, secure tokens, Hardware Security Modules (HSMs), or cloud based Key Management Services.
+- Publishers SHOULD be able to generate multiple signatures for a single artifact.
+- Publisher admins MUST have a mechanism to revoke signatures to indicate they are no longer trusted. 
+- Trust stores MUST be configurable by the deployer admin.
+- Deployer admins MUST be able to configure trusted entities for individual repositories and targets.
+- Deployers MUST be able to validate signatures on any version of an artifact including whether they have been revoked by the publisher.
+- Signature validation MUST be enforceable in air-gapped environments. 
```

Comment:
+1

We might want to state an obvious here. 

> Signature validation MUST be enforceable in air-gapped environments. 

Signature validation MUST be enforceable in air-gapped environments without any external access. 

This accounts for the TSA goals, where the TSA can be verified in the air-gapped environment, as the TSA keys are capable of being copied to the Air-gapped environment.